rhino/src/test/java/org/mozilla/javascript/tests/OptionalChainingOperatorTests.java

```diff
@@ -126,4 +126,27 @@ public void testOptionalChainingOperator() {
                     return null;
                 });
     }
+
+    @Test
+    public void expressionsInOptionalChaining() {
+        Utils.assertWithAllOptimizationLevelsES6(true, "o = {a: true}; o?.['a']");
+        Utils.assertWithAllOptimizationLevelsES6(Undefined.instance, "o = null; o?.['a']");
+        Utils.assertWithAllOptimizationLevelsES6(Undefined.instance, "o = undefined; o?.['a']");
```

Comment:
Could you add a version of this that specifically uses an index (like "o?.[1]")? There is a whole code path in the bytecode that I see we're not going down and I'd feel better if we were.

Reply:
Pushed it, but see my comment above.